prompting/miners/openai_miner.py

```diff
@@ -79,21 +74,45 @@ def __init__(self, config=None):
     def forward(self, synapse: StreamPromptingSynapse) -> Awaitable:
         async def _forward(
             self,
-            message: str,
+            synapse: StreamPromptingSynapse,
             init_time: float,
             timeout_threshold: float,
-            chain: RunnableSequence,
-            chain_formatter: Dict[str, str],
             send: Send,
         ):
             buffer = []
+            accumulated_chunks = []
+            accumulated_chunks_timings = []
+            messages = []
             temp_completion = ""  # for wandb logging
             timeout_reached = False
-
-            try:
-                # Langchain built in streaming. 'astream' also available for async
-                for token in chain.stream(chain_formatter):
-                    buffer.append(token)
+            
+
+            try:                
+                system_prompt_message = [{ 'role': 'system', 'content': self.system_prompt }]
+                synapse_messages = [{'role': role, 'content': message} for role, message in zip(synapse.roles, synapse.messages)]
+                
+                messages = system_prompt_message + synapse_messages
+                
+                start_time = time.time()
+                stream_response = self.model.chat.completions.create(
```

Comment:
I don't think the response actually gets streamed like this. I'm not 100% sure as I'm just doing a superficial review, but I think the way it's being called here you're collecting the whole stream and then sending it off. You'd have to incrementally yield the results as you're receiving the stream

Reply:
That is probably true